gossip/filters/filter_test.go

```diff
@@ -397,3 +396,90 @@ func TestFilter_IndexedLogsReturnsLogsWithTimestampOrError(t *testing.T) {
 		})
 	}
 }
+
+func TestFilter_IndexedLogsReturnsCorrectedTransactionIndexes(t *testing.T) {
+
+	ctrl := gomock.NewController(t)
+	backend := NewMockBackend(ctrl)
+	index := topicsdb.NewMockIndex(ctrl)
+
+	logs := []*types.Log{
+		{
+			BlockNumber: 1,
+			TxHash:      common.HexToHash("0xabc"),
+			Index:       777, // incorrect index
+		},
+		{
+			BlockNumber: 2,
+			TxHash:      common.HexToHash("0x123"),
+			Index:       123, // incorrect index
+		},
+		{
+			BlockNumber: 2,
+			TxHash:      common.Hash{}, // empty hash
+			Index:       123,           // incorrect index
+		},
+	}
+	txs := map[common.Hash]*evmstore.TxPosition{
+		common.HexToHash("0xabc"): {BlockOffset: 7},
+		common.HexToHash("0x123"): {BlockOffset: 1},
+	}
+
+	backend.EXPECT().EvmLogIndex().Return(index).AnyTimes()
+	backend.EXPECT().GetTxPosition(gomock.Any()).
+		DoAndReturn(func(f common.Hash) *evmstore.TxPosition {
+			return txs[f]
+		}).AnyTimes()
+
+	backend.EXPECT().HeaderByNumber(gomock.Any(), gomock.Any()).
+		Return(&evmcore.EvmHeader{
+			Number: big.NewInt(1),
+		}, nil,
+		).AnyTimes()
+	backend.EXPECT().HeaderByHash(gomock.Any(), gomock.Any()).
+		Return(&evmcore.EvmHeader{
+			Number: big.NewInt(1),
+		}, nil,
+		).AnyTimes()
+	backend.EXPECT().GetLogs(gomock.Any(), gomock.Any()).Return([][]*types.Log{logs}, nil).AnyTimes()
+	index.EXPECT().FindInBlocks(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(logs, nil).AnyTimes()
+
+	cases := map[string]*Filter{
+		"filter by block range (unindexed)": {
+			backend: backend,
+			config:  testConfig(),
```

Comment:
testConfig and backend are the same for all 3 cases, please remove it from cases.